R/ids_get.R

```diff
@@ -209,7 +259,24 @@ create_time <- function(start_date, end_date) {
       )
     }
     paste0("YR", seq(start_date, end_date, by = 1))
+  } else if (!is.null(start_date)) {
+    paste0("YR", seq(start_date, latest_year_projections, by = 1))
   } else {
     "all"
   }
 }
+
+filter_post_actual_na <- function(data) {
```

Comment:
I take it you're trying to drop NAs outside the coverage period, but preserve NAs within the coverage period? To avoid hardcoding and continually updating the start and end years, I would suggest:

1) Get the first and last years in the retrieved dataset with a non-NA value, and treat these as the boundaries of the data, then
2) Drop all NAs outside those boundaries.